### PR TITLE
fix Save class

### PR DESCRIPTION
**Pull request by ymirmeddeb**
fixing a bug in the save class

### CODE REVIEW

File: src/main/java/seedu/duke/Save.java [hunk 1]
```diff
@@ -1,37 +1,42 @@
...
     }
 
-
     private static int getUserSaveDirection() {
-        int userDirection;
-        do {
+        int userDirection = -1;
+        while (userDirection == -1) {
             System.out.print("Enter the direction to save the penalty (0, 1, or 2): ");
             String directionString = Ui.IN.nextLine().trim();
```
Comment (runxinghuan) [inline src/main/java/seedu/duke/Save.java:21]: You are reading save int right?

File: src/main/java/seedu/duke/Save.java [hunk 2]
```diff
@@ -1,37 +1,42 @@
...
             }
-        } while (userDirection == -1);
+        }
         return userDirection;
     }
 
-
+    private static int convertToShootDirection(String direction) {
+        try {
+            int directionIndex = Integer.parseInt(direction);
```
Comment (runxinghuan) [inline src/main/java/seedu/duke/Save.java:32]: Can you parse it to int? Because you are trying to parse "save int"

**Review by runxinghuan — COMMENTED** [review]
I want to ask that whether you are able to parse the direction. Also, the command is already read in Ui class, and stored in userInput. You don't need to read it again here.

File: src/main/java/seedu/duke/Save.java (hunk 1 above)
Comment (ymirmeddeb) [reply src/main/java/seedu/duke/Save.java:21]: Yes. The getUserSaveDirection() method prompts the user for input and expects an integer corresponding to the direction to save the penalty. The input is read as a string and then converted into an integer using the convertToShootDirection() method, which checks if the input is within the valid range (0, 1, or 2).

File: src/main/java/seedu/duke/Save.java (hunk 2 above)
Comment (ymirmeddeb) [reply src/main/java/seedu/duke/Save.java:32]: The convertToShootDirection() method is designed to parse the argument for the "save" command, which should be an integer within the range (0, 1, or 2). The string passed to this method is expected to be just the argument part of the input, not the full command. In the case of the "save 1" command, only "1" would be passed to convertToShootDirection(). This is handled by the Parser class, which splits the full input string into the command and argument tokens.

**Pull request closed without merging**